### ISSUE TITLE
Combination of ForAllOtherMembers and ReverseMap

### ISSUE THREAD

**Issue by Devvox93** [AutoMapper/AutoMapper#3427]
Hello, I have a (sadly) nasty situation where information of 2 entities make up 1 logical model.
I have set up mappings to deal with this, as described (simplified) below, so that these entities are combined/merged into 1. This works very good, even with collections, thanks to [AutoMapper.Collection](https://github.com/AutoMapper/AutoMapper.Collection).

In these mappings, I want to use `ReverseMap()` (mainly) for testing purposes. (I have tried setting up a reverse mapping myself using a Tuple as destination, but sadly that gave errors saying I can only map the first level.)
However, I also would like to use `ForAllOtherMembers()` to ignore the members that are not present in the currently mapped entity.
I have searched for and tried several approaches mentioned on StackOverflow and here in the (closed) [issue list](https://github.com/AutoMapper/AutoMapper/issues?q=is%3Aissue+ForAllOtherMembers+and+ReverseMap).
[This](https://github.com/AutoMapper/AutoMapper/issues/1395#issuecomment-335042967) seemed like a dirty hack and the other issues I found only had a brief, rather uninformational response that "it's by design" before being closed (although those issues were simply questions without any detail whatsoever).

The best solution I have now is to use **Memberlist.None**, but with that I lose the validation for other members, so this seems a bit dirty. Therefore I'm looking for a better solution.
Is there a way to combine `ForAllOtherMembers()` and `ReverseMap()`, is there a better way to solve this, or is my current approach (Memberlist.None) the best solution?

And also the question: Why does `ForAllOtherMembers()` return void instead of an IMappingExpression, like (most of) the other methods?

### Source/destination types

```csharp
public class Source1
{
	public int Id1 { get; set; }
	public string InfoInSource1 { get; set; }
// etc.
}

public class Source2
{
	public int Id2 { get; set; }
	public string InfoInSource2 { get; set; }
// etc.
}

public class Destination
{
	public int Id { get; set; }
	public string InfoFromSource1 { get; set; }
	public string InfoFromSource2 { get; set; }
// etc.
}
```

### Mapping configuration

```csharp
CreateMap<Source1, Destination>(MemberList.None)
	.EqualityComparison((src, dest) => src.Id1== dest.Id)
	.ForMember(dest => dest.InfoFromSource1, opt => opt.MapFrom(src => src.InfoInSource1));
// etc.

CreateMap<Source2, Destination>(MemberList.None)
	.EqualityComparison((src, dest) => src.Id2== dest.Id)
	.ForMember(dest => dest.InfoFromSource2, opt => opt.MapFrom(src => src.InfoInSource2));
// etc.

// The tuple mapping (source and destination types match with the classes above, although here they are actually flipped as you can see from the tuple names.):
CreateMap<Destination, (Source1 destination1, Source2 destination2)>()
	.ForMember(dest => dest.destination1, opt => opt.MapFrom(src => src))
	.ForMember(dest => dest.destination2, opt => opt.MapFrom(src => src));
```

### Expected behavior

1. Use `ForAllOtherMembers()` and `ReverseMap()` after.
2. Use `MemberList.Source`
3. Use `MemberList.Destination`
4. Use `Memberlist.None`

### Actual behavior

1. `ForAllOtherMembers()` returns void.
2. Mapping works, but validation fails because destination does not have members for all the source members
3. Mapping works, but validation fails because neither source1 nor source2 have members for all the destination members, since they're combined.
4. Mappings work and validation passes, but that's because the validation is disabled.

### Steps to reproduce

```csharp
// Normal use:
var destination = mapper.Map<Destination>(source1);
mapper.Map(source2, destination); // Occurs if exists and business logic is processed

// Test use:
var destination = Fakers.DestinationFaker.Generate();
var (source1, source2) = testMapper.Map<(Source1, Source2)>(destination);
```

EDIT: After posting this and looking through the code again, I thought of another solution: Make the original mapping work with a tuple as well.
The thing is, there are situations when either source1 or source2 is not needed. Would such a mapping work if one of the source objects in the tuple are null, e.g.: `var destination = mapper.Map<Destination>((source1, null));`.
This would solve my 3rd behavior-issue as then all destination members would be mapped.
But again: isn't there a better solution or approach that I'm overseeing?

**Comment by jbogard** [comment]
I’d say the better solution here is to not use AutoMapper. It’s not designed for scenarios like this.

**Comment by Devvox93** [comment]
I have implemented the solution previously described in my edit (bottom of the main post).
I have now successfully combined both source mapping into a tuple to destination, using `Memberlist.Destination` and `ReverseMap()`.
It works perfectly and if only 1 source member can or has to be mapped, passing null works too.
I've also added convenience mappings of either source-type that use ConstructUsing to call the main tuple-mapping with a null value for the other source variable.

In essence, I don't require the combination of `ForAllOtherMembers()` and `Reversemap()` anymore, although I still wonder why the first does not return an `IMappingExpression` like other methods. It breaks the fluid syntax and prevents the use of `ReverseMap()` (for example).
